src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -164,16 +172,20 @@ class NestedType
                 }
             }
 
-            // Generate into any decl we can find.
+            // If there is a declaration in a non auto-generated file, prefer it.
+            Func<Document,bool> isGeneratedCode = 
+                solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode;
```

Comment:
That's not actually true. Every control dropped is a private field generate by WinForms within the .Designer file.